UI/Panels/Device/MFModulePanel.cs

```diff
@@ -93,20 +88,53 @@ private void DisplayDetails(Board board)
                 buttonDocs.Click += (s, e) => { Process.Start(CreateRedirectTarget(board.Info.Community.Docs)); };
             else
                 buttonDocs.Enabled = false;
-            
+
             if (board.Info.Community?.Support != null)
                 buttonSupport.Click += (s, e) => { Process.Start(CreateRedirectTarget(board.Info.Community.Support)); };
             else
                 buttonSupport.Enabled = false;
 
 
-            
+
             var defaultDeviceConfigFile = board.GetDefaultDeviceConfigFilePath();
-            if (defaultDeviceConfigFile != null && File.Exists(defaultDeviceConfigFile))
+            var defaultDeviceConfigFileExists = defaultDeviceConfigFile != null && File.Exists(defaultDeviceConfigFile);
+            var specificDeviceConfigs = board.Info.DeviceConfigs?.Where(p => File.Exists(board.GetDeviceConfigFilePath(p.File)));
+            var atLeastOneSpecificConfigExists = specificDeviceConfigs.Count() > 0;
```

Comment:
Feels like this logic should be in the class itself not here. This code should just be able to call `board.Info.DeviceConfigs` and only get back ones that actually exist instead of having to worry about it. Same for calculating an `atLeastOneSpecificConfigExists` property, make it a getter-based property on the class.